app/controllers/mixins/actions/vm_actions/reconfigure.rb

```diff
@@ -353,11 +355,6 @@ def reconfigure_handle_submit_button
             url = previous_breadcrumb_url.split('/')
             javascript_redirect(:controller => url[1], :action => url[2])
           end
-
-          if @flash_array
```

Comment:
This bit is surprising, why would you remove a flash message?

Reply:
Thanks for asking. This flash rendering breaks redirect as far as I've tested because it happend AFTER redirecting to other page. See the condition in L352:

```ruby
if ...
   javascript_redirect(...)
else
   javascript_redirect(...)
end
```

There is no way we can arrive to L357 without callind redirect first. Or am I missing something? Looking forward to your answer, I'm curious how this can work for other providers. Most probably it doesn't :)

Reply:
Aah... I think you're right.. A few lines above, there is a `flash_to_session` call which saves the flash messages for later.

Since both branches do `javascript_redirect`, unconditionally, I agree this is broken.
And since `flash_to_session` I think we can expect this won't hide any flash messages that would otherwise appear.

:+1: